dask_cuda/benchmarks/local_merge.py

```diff
@@ -0,0 +1,216 @@
+        #
+        # "payload" column is a random permutation of the chunk_size
+
+        start = local_size * i_chunk
+        stop = start + local_size
+        df = cudf.DataFrame(
+            {
+                "key": cupy.arange(start, stop=stop, dtype="int64"),
+                "payload": cupy.random.permutation(
+                    cupy.arange(local_size, dtype="int64")
+                ),
+            }
+        )
+    else:
+        # Other dataframe
+        #
+        # "key" column matches values from the build dataframe
+        # for a fraction (`frac_match`) of the entries. The matching
+        # entries are perfectly balanced across each partition of the
+        # "base" dataframe.
+        #
+        # "payload" column is a random permutation of the chunk_size
+
+        # Step 1. Choose values that DO match
+        sub_local_size = local_size // num_chunks
+        sub_local_size_use = int(sub_local_size * frac_match)
+        arrays = []
+        for i in range(num_chunks):
+            bgn = (local_size * i) + (sub_local_size * i_chunk)
+            end = bgn + sub_local_size
+            ar = cupy.arange(bgn, stop=end, dtype="int64")
+            arrays.append(cupy.random.permutation(ar)[:sub_local_size_use])
+        key_array_match = cupy.concatenate(tuple(arrays), axis=0)
+
+        # Step 2. Add values that DON'T match
+        missing_size = local_size - key_array_match.shape[0]
+        start = local_size * num_chunks + local_size * i_chunk
+        stop = start + missing_size
+        key_array_no_match = cupy.arange(start, stop=stop, dtype="int64")
+
+        # Step 3. Combine and create the fineal dataframe chunk (dask_cudf partition)
+        key_array_combine = cupy.concatenate(
+            (key_array_match, key_array_no_match), axis=0
+        )
+        df = cudf.DataFrame(
+            {
+                "key": cupy.random.permutation(key_array_combine),
+                "payload": cupy.random.permutation(
+                    cupy.arange(local_size, dtype="int64")
+                ),
+            }
+        )
+    return df
+
+
+def get_random_ddf(chunk_size, num_chunks, frac_match, chunk_type):
+
+    parts = [chunk_size for i in range(num_chunks)]
+    meta = generate_chunk(0, 4, None, None, None)
+    divisions = [None] * (len(parts) + 1)
+
+    name = "generate-data-" + tokenize(chunk_size, num_chunks, frac_match, chunk_type)
+
+    graph = {
+        (name, i): (generate_chunk, i, part, len(parts), chunk_type, frac_match)
+        for i, part in enumerate(parts)
+    }
+
+    return new_dd_object(graph, name, meta, divisions)
+
+
+def main(args):
+    n_workers = len(args.devs.split(","))
+
+    # Set up workers on the local machine
+    cluster = LocalCUDACluster(
+        protocol=args.protocol, n_workers=n_workers, CUDA_VISIBLE_DEVICES=args.devs,
+    )
+    client = Client(cluster)
+
+    if args.no_pool_allocator:
+        client.run(lambda: cudf.set_allocator("default", pool=False))
```

Comment:
Small style suggestion

```suggestion
        client.run(cudf.set_allocator, "default", pool=False)
```